### ISSUE TITLE
False positive with RUF009 using NewType, Final and dataclass

### ISSUE THREAD

**Issue by inkhey** [astral-sh/ruff#6447]
Hello :wave: , I think I found a false positive on `RUF009` rule. 

test_ruff.py:
```python
from dataclasses import dataclass
from typing import Final, NewType

State = NewType("State", str)


@dataclass
class System:
    state: Final[State] = State("Enabled")


@dataclass
class System2:
    state: Final[str] = "Enabled"
```
version:
```bash
❯ ruff --version
ruff 0.0.283
```
bug.toml:
```toml
select = ["RUF"]
```
output:
```bash
❯ ruff check --config=bug.toml test_ruff.py 
test_ruff.py:9:27: RUF009 Do not perform function call `State` in dataclass defaults
Found 1 error.
```

**Comment by zanieb** [comment]
Thanks for the report! We could special case this by looking up the binding for `State` and checking it it's a `NewType` declaration. However, this will not work across files, with aliased assignments, etc. because we do not have type inference.
